src/EventListener/AuditLogListener.php

```diff
@@ -0,0 +1,146 @@
+        'popup_new_dashboard_unpublish',
+        'popup_new_duplicate',
+        'popup_new_lead_delete',
+        'lightbox_create',
+        'lightbox_rename',
+        'lightbox_delete',
+        'lightbox_duplicate',
+        'lightbox_save',
+        'account_billing_info_change',
+        'account_cancel_subscription',
+        'account_add_font',
+        'account_delete_font',
+        'account_export_lead',
+        'account_archive_lead',
+        'account_delete_lead',
+        'account_restore_lead',
+        'account_enable_feature',
+        'account_disable_feature',
+        'account_create_token',
+        'account_delete_token',
+        'account_create_user',
+        'account_delete_user',
+        'account_add_domain',
+        'account_domain_assign_landing',
+        'account_delete_domain',
+        'account_share_domain',
+        'account_change_timezone',
+        'account_change_package',
+        'account_change_default_landing_language',
+        'agency_edit_user',
+        'agency_remove_user',
+        'agency_give_user_access_to_subaccount',
+        'agency_remove_user_access_to_subaccount',
+        'agency_add_admin',
+        'agency_edit_admin',
+        'agency_delete_admin',
+        'agency_add_subaccount',
+        'agency_activate_subaccount',
+        'agency_deactivate_subaccount',
+        'agency_delete_subaccount',
+        'agency_edit_subaccount',
+        'agency_gallery_create_folder',
+        'agency_gallery_rename_folder',
+        'agency_gallery_delete_folder',
+        'agency_gallery_delete_picture',
+        'agency_gallery_upload_picture',
+        'agency_gallery_moved_picture',
+        'agency_create_template',
+        'agency_restore_template',
+        'agency_archive_template',
+        'agency_duplicate_template',
+        'agency_rename_template',
+        'agency_save_template',
+        'agency_duplicate_landing_to_sub_account',
+        'change_user_email',
+        'log_in',
+        'published',
+        'save',
+        'sms_sent',
+        'phone_verified',
+        'landing_group_page_assigned',
+        'landing_group_page_dismissed',
+        'landing_move',
+        'unsplash_image_import',
+        'account_resource_limit_created',
+        'account_resource_limit_edited',
+        'account_resource_limit_deleted',
+        'account_resource_limit_attached',
+        'account_resource_limit_detached',
+    ];
+
+    private EventDataStore $auditLogEventData;
+    private AuditLogClient $auditLogClient;
+
+    public function __construct(EventDataStore $auditLogEventData, AuditLogClient $auditLogClient)
+    {
+        $this->auditLogEventData = $auditLogEventData;
+        $this->auditLogClient = $auditLogClient;
+    }
+
+    public function onEvent(Event $event): void
+    {
+        if (in_array((string) $event->getName(), self::AUDITLOG_EVENTS, true)) {
+            $this->auditLogClient->store(
+                new AuditLogEvent(
```

Comment:
Nie wolisz sobie rozszerzyć obiektu Event, albo go udekorować? - skoro tyle danych jest współdzielonych.

Będziesz miał mniej getterów, które tutaj no wyglądają słabo.

Reply:
moge zrobic extend Event, nie zrobilem bo jest jako final wiec uznalem, że tak chciałeś :P takem uczynię

Reply:
co do getterów to niech zostaną bo tylko tutaj dane przepychamy, tak jest najwygodniej

Reply:
jak chcesz mieć bez getterów to obiekt AuditLogEvent powienien klienta do AuditLoga przyjmować i do niego wsadzać co trzeba, ale tutaj chyba w enkapsulację sie nie bawimy patrząc na to co jest do tej pory